crates/rpc/rpc/src/eth/api/mod.rs

```diff
@@ -341,6 +357,11 @@ where
                 expires_at: now + Duration::from_secs(3),
             });
 
+            let receipts = this.cache().get_receipts(pending_block.hash).await?.unwrap_or_default();
+            let pending_block_clone = pending_block.block.clone();
+            this.local_pending_block_sender()
+                .send_modify(|v| *v = Some((pending_block_clone, receipts.to_vec())));
```

Comment:
this doesn't quite work, because the cache is unaware of this block.

we'd need to also store receipts here:

https://github.com/paradigmxyz/reth/blob/021d236f8c7a067cfbcc4ecd7ff77b6a72adc1cb/crates/rpc/rpc/src/eth/api/pending_block.rs#L359-L359

Reply:
MMM
ok! I was thinking about it but decided to try cache

Reply:
add, but need to think about how to init SealedBlockWithSender and BlockWithSender in places in which we don't have receipts
somewhere I get them, somewhere I init with empty vector

Reply:
Created a new type SealedBlockWithSendersAndReceipts